### DISCUSSION TITLE
Training model using training, validation and test set

### DISCUSSION THREAD

**Discussion by minhhb9x** [ultralytics/ultralytics#2135] [Q&A] [answered]
Assuming that I have a yaml file with directories to training, validation and test set like this:
 ```
path: ../datasets/coco128  
train: images/train2017  
val: images/val2017  
test:  images/test2017  
```
I don't know how to run on the validation set on each epoch during the training process and how to run on the test set after training process is done. Can sb help me, pls?

**Comment by Laughing-q** [comment, marked as answer]
@minhhotboy9x the evaluation on val set is running during training by default, you don't need to worry about that.
To evaluate test set after training, you can run the following command:
```bash
yolo detect val model=best.pt data=your_data.yaml split=test
```

**Reply by minhhb9x** [reply]
Thank you, I will try this

**Reply by glenn-jocher** [reply]
@minhhotboy9x you're welcome! If you have any more questions or need further assistance, feel free to ask. Happy training! 😊

**Comment by ookinim** [comment]
> @minhhotboy9x the evaluation on val set is running during training by default, you don't need to worry about that. To evaluate test set after training, you can run the following command:
> 
> ```shell
> yolo detect val model=best.pt data=your_data.yaml split=test
> ```

Please suggest which best.pt to use for resume training with the last.pt is done? as new train folder (after resume model) does not contain the best.pt
